modules/private_dns_cname_record/variables.tf

```
@@ -1,10 +1,16 @@
 variable "cname" {
   type        = string
-  description = "The CNAME record value."
+  description = "CNAME (FQDN), strict DNS rules; trailing dot optional."
 
   validation {
-    condition     = can(regex("^[a-zA-Z0-9][a-zA-Z0-9-_.]*[a-zA-Z0-9]$", var.cname)) && length(var.cname) <= 253
-    error_message = "The cname must be a valid CNAME and not exceed 253 characters."
+    condition = can(
+      regex(
+        "^(([a-zA-Z0-9_]|[a-zA-Z0-9_][a-zA-Z0-9_\-]*[a-zA-Z0-9_])\.)*([A-Za-z0-9_]|[A-Za-z0-9_][A-Za-z0-9_\-]*[A-Za-z0-9_])(\.?)$",
```

```suggestion
        "^(([a-zA-Z0-9_]|[a-zA-Z0-9_][a-zA-Z0-9_\\-]*[a-zA-Z0-9_])\\.)*([A-Za-z0-9_]|[A-Za-z0-9_][A-Za-z0-9_\\-]*[A-Za-z0-9_])(\\.?)$",
```

Sorry, seems we need to escape the escape in Terraform...